pkg/manifests/tls.go

```diff
@@ -36,94 +39,150 @@ func needsNewCert(notBefore, notAfter time.Time, now func() time.Time) bool {
 	return now().After(latestTime)
 }
 
-// This method creates a central secret containing GRPC TLS key material
-// if the given secret is nil
-// or updates it if the CA present in the given secret is about to expire.
+func (f *Factory) GRPCSecret() (*v1.Secret, error) {
```

Comment:
I wonder if we can move this to the manifests.go now, that is purely just manifests generation? Not urgent to address just as we have this pattern already.

Reply:
Yeah, i was back and forth on this one. It could be considered the _first_ step to untangle `manifests.go` which is too big already. I would leave it here and consequently cut `manifests.go` in separate files as a follow-up.

Reply:
But the test for it is in `pkg/manifests/manifests_test.go `, I would then either move this function there or move the test to tls_test.go?

Reply:
let's move the test to `tls_test.go` indeed and set a first cutting precedence :+1: